src/plugins/discover/public/plugin.tsx

```diff
@@ -18,6 +18,7 @@ import {
   Plugin,
   PluginInitializerContext,
 } from 'kibana/public';
+import { isFilterPinned } from '@kbn/es-query';
```

Comment:
dear @elastic/kibana-operations , not 100% sure but I think this adds +27.5KB to the page load bundle, could we improve that? thx

Reply:
1. import it asynchronously: Looks like this function is used in the definition of the `KbnUrlTracker` which requires url trackers to be defined even when the app isn't in use, which doesn't seem to make sense to me. If that wasn't necessary then you could async import all the KbnUrlTracker logic when discover is mounted for the first time... not sure how flexible that code is though
2. use a submodule: @mistic needed to remove all the sub-module stuff we've tried previously as we moved to split `@types/*` packages for TS performance, so I don't think this is a useful idea anymore but maybe @mistic could correct us.
3. split up this package: if it's reasonable for people to import this package for a single function we might want to split up `@kbn/es-query` so there are smaller packages people can import for specific functionality
4. convert filters to instances of some class that carry helper methods around like this so they don't need to be imported everywhere they are used. Instead you could do write something like `state.filters?.filter(f => f.isPinned())`

Reply:
Thx spencer, think I'll go for 1. 👍

Reply:
note that I didn't import all of the KbnUrlTracker logic, this caused troubles, but just importing the part used for pinning async worked